README.md

```diff
@@ -1,74 +1,52 @@
+  plugins: [require.resolve('prettier-plugin-astro')],
+  overrides: [
+    {
+      files: '*.astro',
+      options: {
+        parser: 'astro',
+      },
+    },
+  ],
+};
 ```
 
-### pnpm support
+The `require.resolve` call can alternatively be changed to a direct path, like such: `plugins: ["./node_modules/prettier-plugin-astro"]` for usage inside a non-JS config file.
 
-Due to [an upstream issue in Prettier](https://github.com/prettier/prettier/issues/8056), the `plugin-search-dir` parameter should be set to the current directory when using pnpm or Prettier won't be able to find the plugin automatically
-
-```shell
-prettier -w --plugin-search-dir=. .
-```
+To customize formatting behavior, see the [Configuration](#configuration) section below.
 
 ## Using in VS Code
 
-First install the [VS Code Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) and add the following settings to your VS Code configuration so VS Code is aware that Prettier can be used for Astro files:
+> **Note**
+> If you're using [Astro's VS Code extension](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode), Prettier and this plugin are already included. Only follow the guide below to format using Prettier's official extension.
 
-```json
-{
-  "prettier.documentSelectors": ["**/*.astro"]
-}
-```
-
-Additionally, you should set Prettier as the default formatter for Astro files or VS Code will ask you to choose a formatter everytime you format since the Astro VS Code extension also includes a formatter for Astro files:
+First install the [VS Code Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) and add the following settings to your VS Code configuration so VS Code is aware that Prettier can be used for Astro files and set it as the default formatter for Astro files:
 
 ```json
 {
+  "prettier.documentSelectors": ["**/*.astro"],
   "[astro]": {
     "editor.defaultFormatter": "esbenp.prettier-vscode"
   }
 }
 ```
 
-When submitting issues about formatting in VS Code, first make sure you're actually using Prettier to format your files and not the Astro VS Code extension included formatter
-
-### pnpm support
-
-Due to an upstream issue, Prettier inside VS Code isn't able to automatically infer the right parser to use for Astro files when using pnpm
-
-As such, add the following settings to your `.prettierrc.cjs` config file:
-
-```js
-module.exports = {
-  plugins: [require.resolve('prettier-plugin-astro')],
-  overrides: [
-    {
-      files: '*.astro',
-      options: {
-        parser: 'astro',
-      },
-    },
-  ],
-};
-```
-
-The `require.resolve` call can alternatively be changed to a direct path, like such: `plugins: ["./node_modules/prettier-plugin-astro"]` for usage inside a non-JS config file
+When submitting issues about formatting in VS Code, please specify if you're using the Astro extension or the Prettier extension to format your files.
```

Comment:
```suggestion
 The settings above ensure that VS Code is aware that Prettier can be used for Astro files, and sets Prettier as the default formatter for Astro files.

### Reporting issues

When submitting issues about formatting your `.astro` files in VS Code, please specify which extension you are using to format your files: Astro's own extension or the Prettier extension.
```
2 things:
- I put the explanation of the settings underneath, so that the instructions to follow were a simple one-liner (and, people can skip WHY this works and what it's doing if they want)
- I made a separate heading for reporting issues because people have been told to skip this section if they're using the Astro extension, so they might not even see this. I made this a L3 heading under the VS code section, since it is related to VS Code. But, if you felt like you wanted to call more attention to this with an L2 heading, I think that would be fine too.